File: packages/types/src/instance/workspace.ts
/**
 * Copyright (c) 2023-present Plane Software, Inc. and contributors
 * SPDX-License-Identifier: AGPL-3.0-only
 * See the LICENSE file for details.
 */

/**
 * Instance-wide workspace-creation policy keys (distinct from `../workspace.ts`
 * which models individual workspace entities); read via
 * `IInstanceConfig.is_workspace_creation_disabled` in `apps/web`.
 */

/**
 * Workspace-creation policy key — `DISABLE_WORKSPACE_CREATION="1"` rejects
 * non-admin create requests at the API and hides the create-workspace UI;
 * mirrored into `IInstanceConfig.is_workspace_creation_disabled` at boot.
 */
export type TInstanceWorkspaceConfigurationKeys = "DISABLE_WORKSPACE_CREATION";

File: apps/api/plane/api/serializers/workspace.py
# Copyright (c) 2023-present Plane Software, Inc. and contributors
# SPDX-License-Identifier: AGPL-3.0-only
# See the LICENSE file for details.
"""Lightweight workspace serializer for the ``/api/v1/`` API surface.

Embedded by sibling serializers in :mod:`plane.api.serializers` via the
``?expand=workspace`` mechanism wired in
:class:`plane.api.serializers.base.BaseSerializer`. All fields are read-only
because the API never accepts workspace data through this serializer — it
is purely a read projection used to attach ``{id, slug, name}`` to enclosing
records.
"""

# Module imports
from plane.db.models import Workspace
from .base import BaseSerializer


class WorkspaceLiteSerializer(BaseSerializer):
    """
    Lightweight workspace serializer for minimal data transfer.

    Provides essential workspace identifiers including name, slug, and ID
    optimized for navigation, references, and performance-critical operations.
    """

    class Meta:
        """DRF metadata: serialize ``Workspace`` exposing read-only ``id``, ``slug``, and ``name``."""

        model = Workspace
        fields = ["name", "slug", "id"]
        read_only_fields = fields
